coap-gateway/service/service.go

```diff
@@ -173,7 +215,33 @@ func New(ctx context.Context, config Config, logger log.Logger) (*Service, error
 			return nil, fmt.Errorf("cannot create tls cert manager: %w", err)
 		}
 		nats.AddCloseFunc(coapsTLS.Close)
-		l, err := net.NewTLSListener("tcp", config.APIs.COAP.Addr, coapsTLS.GetTLSConfig())
+		tlsCfgForClient := coapsTLS.GetTLSConfig()
+		var tlsCfg tls.Config
```

Comment:
Don't you want to add this new part to a different func ? This `New()` is already pretty huge and unreadable.

Reply:
moved to separate func